src/v/cluster/partition_manager.cc

```diff
@@ -121,8 +121,12 @@ ss::future<consensus_ptr> partition_manager::manage(
   std::optional<xshard_transfer_state> xst_state,
   std::optional<remote_topic_properties> rtp,
   std::optional<cloud_storage_clients::bucket_name> read_replica_bucket,
-  std::optional<cloud_storage::remote_label> remote_label,
-  std::optional<model::topic_namespace> topic_namespace_override) {
+  const topic_configuration* topic_cfg) {
```

Comment:
nit: why not reference to avoid the nullptr check?

Reply:
Including an extra dep in a .cc seems fine?

Reply:
Yeah maybe. We'll also need to include it where we call it with an empty optional. Or rather use an `std::optional<std::reference_wrapper<topic_configuration>>`.

Is there any reason we might prefer it to a raw pointer? It is lengthier to type and read, adds memory overhead, is similarly (un)safe. Possibly hints the "un-owning-ness" a bit better, but I thought in 2025 everyone uses smart pointers where it is meant to own. Sounds like a topic for #cpplang. There was a similar discussion regarding an owning one https://redpandadata.slack.com/archives/C03RTPCKS69/p1740080577471999 with little consensus.